app/src/main/java/audiomodem/AutoBroadcaster.java

```diff
@@ -0,0 +1,54 @@
+package audiomodem;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.os.Looper;
+import android.util.Log;
+
+import utils.ModemCotUtility;
+
+import com.atakmap.android.cot_utility.plugin.PluginLifecycle;
+import com.atakmap.android.maps.MapView;
+
+public class AutoBroadcaster implements Runnable {
+    private ModemCotUtility modemCotUtility;
+    private MapView mapView;
+    private Thread thread;
+    private boolean running;
+ 
+    final static String TAG = "AutoBroadcaster";
+
+    public AutoBroadcaster(ModemCotUtility modemCotUtility, MapView mapView){
+        this.modemCotUtility = modemCotUtility;
+        this.mapView = mapView;
+        this.running = true;
+        this.thread = new Thread(this);
+    }
+    @Override
+    public void run() {
+        Looper.prepare();
+
+        int delay;
+        SharedPreferences sharedPref = PluginLifecycle.activity.getSharedPreferences("hammer-prefs", Context.MODE_PRIVATE);
+
+        try {
+            while(running) {
+                modemCotUtility.stopListener();
+                modemCotUtility.sendCoT(mapView.getSelfMarker());
+                delay = Integer.parseInt(sharedPref.getString("autoBroadcastInterval", "60000")); // default 60 seconds delay

```

Comment:
60 seconds feels too short.   If you wanted to keep 50% channel availability, and each update takes 5s (somewhat arbitrary value, but seems about enough for a single update based on my testing) then you'll have (60s * 0.5) / 5s = 6 simultaneous users (max) able to use this.

Understanding that this is a naive implementation, but upping this to a more conservative default (say 3-5m) and then allowing the user to drop this down to something more frequent seems like a better philosophy.

Reply:
Sounds reasonable.